stdlib/public/runtime/Metadata.cpp

```diff
@@ -3015,3 +3015,15 @@ void MetadataAllocator::Deallocate(const void *allocation, size_t size) {
 void *swift::allocateMetadata(size_t size, size_t alignment) {
   return MetadataAllocator().Allocate(size, alignment);
 }
+
+template<>
+bool Metadata::satisfiesClassConstraint() const {
+  // existential types marked with @objc satisfy class requirement.
+  if (isAnyExistentialType()) {
+    auto existential = reinterpret_cast<const ExistentialTypeMetadata *>(this);
```

Comment:
You can use LLVM's casting machinery in the runtime, e.g.

```
if (auto existential = dyn_cast<ExistentialTypeMetadata>(this)) { ... }
```

Reply:
Done!